modal/_utils/grpc_utils.py

```diff
@@ -307,6 +360,39 @@ async def _retry_transient_errors(
             with suppress_tb_frames(1):
                 return await fn_callable(req, metadata=attempt_metadata, timeout=timeout)
         except (StreamTerminatedError, GRPCError, OSError, asyncio.TimeoutError, AttributeError) as exc:
+            # Server side instruction for retries
+            if isinstance(exc, GRPCError) and (server_retry_policy := get_server_retry_policy(exc)):
+                server_delay = server_retry_policy.retry_after_secs
+
+                # When `total_deadline` is not defined and server gives instruction for retries, then use
+                # `max_throttle_wait`.
+                if total_deadline is None and (max_throttle_wait := config.get("max_throttle_wait")):
+                    total_deadline = t0 + max_throttle_wait
+
+                final_attempt = (
+                    total_deadline is not None
```

Comment:
Should we remove the two cases I could find where we're apparently unnecessarily passing `total_timeout`? I.e. it has a functional role in the Queue operations because we're setting `max_retries=None` there, but it's passed in `FunctionGetCurrentStats` and `ContainerExecPutInputRequest` in a way that doesn't currently have any effect but might if we add throttling to those endpoints.